python/src/aac/lang/lsp/server.py

```diff
@@ -0,0 +1,84 @@
+"""The Architecture-as-Code Language Server."""
+
+import logging
+from typing import Optional
+from pygls.server import LanguageServer
+import pygls.lsp.methods as methods
+
+from pygls.lsp import (
+    CompletionOptions,
+    CompletionParams,
+    Hover,
+    HoverParams,
+    InitializeParams,
+    InitializeResult,
+    ServerCapabilities,
+)
+
+from aac.lang.active_context_lifecycle_manager import get_active_context
+from aac.lang.language_context import LanguageContext
+from aac.lang.lsp.code_completion_provider import CodeCompletionProvider
+
+
+LOGGER: Optional[logging.Logger] = None
+SERVER: Optional[LanguageServer] = None
+ACTIVE_CONTEXT: Optional[LanguageContext] = None
+COMPLETION_PROVIDER: Optional[CodeCompletionProvider] = None
+
+
+def start_lsp(dev: bool = False):
```

Comment:
This was moved to the `aac.plugins.start_lsp.start_lsp_impl` module in my previous PR (#286) along with a few other changes. We should sync those changes with what's in here. Personally, I'm still in favor of putting the `SERVER.start_<whatever>()` call in the plugin and we could make this into an `prepare_lsp()` function, or something, since it will be similar for all the `start_lsp*` commands.

Reply:
Another thing that was changed was that I removed the `dev` parameter.

Reply:
We discussed converting this to a class to manage those instances instead of globals, and reworking the start command. I'll address these in #227